bears/general/AnnotationBear.py

```diff
@@ -28,104 +27,225 @@ def run(self, filename, file, language: str, coalang_dir: str = None):
+                               string_delimiters,
+                               multiline_string_delimiters,
+                               comment_delimiter,
+                               multiline_comment_delimiters):
         """
-        Gives all positions of annotations which have a start and end.
+        Find ranges of all annotations.
 
-        :param filename: Name of the file on which the bear is running.
-        :param file:     Contents of the file in the form of tuple of lines.
-        :param annot:    A dict containing start of annotation as key and end of
-                         annotation as value.
-        :return:         A set of SourceRanges giving the range of annotation.
+        :param file:
+            A tuple of strings, with each string being a line in the file.
+        :param filename:
+            The name of the file.
+        :param string_delimiters:
+            A dictionary containing the various ways to  define single-line
+            strings in a language.
+        :param multiline_string_delimiters:
+            A dictionary contatining the various ways to define multi-line
+            strings in a language.
+        :param comment_delimiter:
+            A dictionary contatining the various ways to define single-line
+            comments in a language.
+        :param multiline_comment_delimiters:
+            A dictionary contatining the various ways to define multi-line
+            comments in a language.
+        :return:
+            Two tuples first contatining a tuple of strings, the second a tuple
+            of comments.
         """
         text = ''.join(file)
-        found_pos = set()
-        for annot_type in annot:
-            found_pos.update(unescaped_search_in_between(
-                                           annot_type, annot[annot_type], text))
-        if found_pos:
-            found_pos = set(SourceRange.from_absolute_position(
-                                filename,
-                                AbsolutePosition(file, position.begin.range[0]),
-                                AbsolutePosition(
-                                    file, position.end.range[1] - 1))
-                            for position in found_pos)
-        return found_pos
+        strings_range = []
+        comments_range = []
+        position = 0
+        while position <= len(text):
+
+            def get_new_position():
+                for string in multiline_string_delimiters.keys():
+                    if text[position:].startswith(string):
+                        _range, end_position = self.get_multiline(
+                            file,
+                            filename,
+                            text,
+                            string,
+                            multiline_string_delimiters[string],
+                            position)
+                        strings_range.append(_range)
+                        return end_position + 1
```

Comment:
maybe put each block in an own function? like on function for multiline strings, one for strings,one for comments...
This one is pretty massive (and nested). I get that you are kind of doing that with `get_multiline` and `get_singleline_strings` but this function is still huge. Maybe you can even simplify to one or two sub-functions called with different params since these block look really similar.